persistence/src/commonMain/kotlin/com/wire/kalium/persistence/dao/UserDAO.kt

```diff
@@ -54,6 +54,25 @@ data class UserEntity(
     val deleted: Boolean
 )
 
+data class UserEntityWithTeam(
+    val id: QualifiedIDEntity,
+    val name: String?,
+    val handle: String?,
+    val email: String?,
+    val phone: String?,
+    val accentId: Int,
+    val team: String?,
+    val connectionStatus: ConnectionEntity.State = ConnectionEntity.State.NOT_CONNECTED,
+    val previewAssetId: UserAssetIdEntity?,
+    val completeAssetId: UserAssetIdEntity?,
+    val availabilityStatus: UserAvailabilityStatusEntity,
+    val userType: UserTypeEntity,
+    val botService: BotEntity?,
+    val deleted: Boolean,
+    val teamName: String?,
+    val teamIcon: String?
+)
```

Comment:
Why `UserEntityWithTeam`?

What about something like `Pair<UserEntity, TeamEntity>` or similar ?

Then we can more easily map `UserEntity` to `User` and `TeamEntity` to `Team` for example, reusing the current mapping functions if needed.